### ISSUE TITLE
IDE crashes when making or editing `Invalid_Aggregate_Column .Count_Distinct` node.

### ISSUE THREAD

**Issue by farmaazon** [enso-org/enso#6117]
When I try to add or edit `Invalid_Aggregate_Column .Count_Distinct` (don't ask), then at some point IDE crashes with panic:

```
panicked at 'size is zero', /rustc/1e4f90061cc4bc566f99ab21b1f101182b10cf0c/library/core/src/slice/mod.rs:787:44

Stack:

Error
    at imports.wbg.__wbg_new_abda76e883ba8a5f (eval at <anonymous> (http://localhost:8080/index.js:74:5709), <anonymous>:17609:17)
    at console_error_panic_hook::format_panic::hb22caa0fd463d2f7 (http://localhost:8080/pkg-opt.wasm:wasm-function[19602]:0x1670a45)
    at enso_web::report_panic::h68568ba56aeeba21 (http://localhost:8080/pkg-opt.wasm:wasm-function[40628]:0x1b9732c)
    at core::ops::function::Fn::call::h24b9848d998459a0 (http://localhost:8080/pkg-opt.wasm:wasm-function[62637]:0x1d65cf8)
    at std::panicking::rust_panic_with_hook::hdc43d6b15c221d2e (http://localhost:8080/pkg-opt.wasm:wasm-function[33723]:0x1a4509c)
    at std::panicking::begin_panic_handler::{{closure}}::h138701bc4f366007 (http://localhost:8080/pkg-opt.wasm:wasm-function[44329]:0x1c371e6)
    at std::sys_common::backtrace::__rust_end_short_backtrace::h6a334029ca520246 (http://localhost:8080/pkg-opt.wasm:wasm-function[58203]:0x1d4fe06)
    at rust_begin_unwind (http://localhost:8080/pkg-opt.wasm:wasm-function[49140]:0x1cc1277)
    at core::panicking::panic_fmt::hfb8e3b4832705da8 (http://localhost:8080/pkg-opt.wasm:wasm-function[55602]:0x1d2adc7)
    at core::panicking::panic_display::hf1bfbeab90bf4dd7 (http://localhost:8080/pkg-opt.wasm:wasm-function[49566]:0x1ccac16)
    at core::panicking::panic_str::hacb805faa711b6b7 (http://localhost:8080/pkg-opt.wasm:wasm-function[58471]:0x1d52dc7)
    at core::option::expect_failed::h46dbf58c483f9509 (http://localhost:8080/pkg-opt.wasm:wasm-function[60716]:0x1d601d8)
    at enso_gui::controller::searcher::input::Input::after_inserting_suggestion::hb39a97c0e73d3773 (http://localhost:8080/pkg-opt.wasm:wasm-function[960]:0x5b6d10)
    at enso_gui::controller::searcher::Searcher::preview_suggestion::h6de9ac88220a0068 (http://localhost:8080/pkg-opt.wasm:wasm-function[908]:0x57462f)
    at enso_gui::presenter::searcher::Model::suggestion_selected::h87efa7ad5f8eed68 (http://localhost:8080/pkg-opt.wasm:wasm-function[3572]:0xbf0a82)
    at <enso_frp::stream::WeakNode<Def> as enso_frp::stream::WeakEventConsumer<T>>::on_event_if_exists::hfacecdc40050d289 (http://localhost:8080/pkg-opt.wasm:wasm-function[31418]:0x19c5558)
    at <enso_frp::stream::OwnedStream<Out> as enso_frp::stream::EventEmitter>::emit_event::hfe24dc111e8ad59a (http://localhost:8080/pkg-opt.wasm:wasm-function[3895]:0xc76335)
    at <enso_frp::stream::WeakNode<Def> as enso_frp::stream::WeakEventConsumer<T>>::on_event_if_exists::h20f68dcf4e25eef4 (http://localhost:8080/pkg-opt.wasm:wasm-function[27251]:0x18be73f)
    at <enso_frp::stream::OwnedStream<Out> as enso_frp::stream::EventEmitter>::emit_event::hb33fd182a6690ac1 (http://localhost:8080/pkg-opt.wasm:wasm-function[3592]:0xbf9257)
    at <enso_frp::stream::WeakNode<Def> as enso_frp::stream::WeakEventConsumer<T>>::on_event_if_exists::h99aad3875007cfc4 (http://localhost:8080/pkg-opt.wasm:wasm-function[27182]:0x18b988c)
    at <enso_frp::stream::OwnedStream<Out> as enso_frp::stream::EventEmitter>::emit_event::hb33fd182a6690ac1 (http://localhost:8080/pkg-opt.wasm:wasm-function[3592]:0xbf9257)
    at <enso_frp::stream::WeakNode<Def> as enso_frp::stream::WeakEventConsumer<T>>::on_event_if_exists::hadd017c45dcf7018 (http://localhost:8080/pkg-opt.wasm:wasm-function[35205]:0x1a936e3)
    at <enso_frp::stream::OwnedStream<Out> as enso_frp::stream::EventEmitter>::emit_event::hb33fd182a6690ac1 (http://localhost:8080/pkg-opt.wasm:wasm-function[3592]:0xbf929b)
    at <enso_frp::stream::WeakNode<Def> as enso_frp::stream::WeakEventConsumer<T>>::on_event_if_exists::he5fa1969dca539df (http://localhost:8080/pkg-opt.wasm:wasm-function[43807]:0x1c2026b)
    at <enso_frp::stream::OwnedStream<Out> as enso_frp::stream::EventEmitter>::emit_event::hb33fd182a6690ac1 (http://localhost:8080/pkg-opt.wasm:wasm-function[3592]:0xbf9257)
    at <enso_frp::stream::WeakNode<Def> as enso_frp::stream::WeakEventConsumer<T>>::on_event_if_exists::h0e71fe58ec888d3d (http://localhost:8080/pkg-opt.wasm:wasm-function[24101]:0x17d48b9)
    at <enso_frp::stream::OwnedStream<Out> as enso_frp::stream::EventEmitter>::emit_event::he9e3a4940042f77f (http://localhost:8080/pkg-opt.wasm:wasm-function[3654]:0xc130f8)
    at <enso_frp::stream::WeakNode<Def> as enso_frp::stream::WeakEventConsumer<T>>::on_event_if_exists::h82207a6b7311f0a9 (http://localhost:8080/pkg-opt.wasm:wasm-function[28603]:0x191a21f)
    at <enso_frp::stream::OwnedStream<Out> as enso_frp::stream::EventEmitter>::emit_event::he9e3a4940042f77f (http://localhost:8080/pkg-opt.wasm:wasm-function[3654]:0xc13088)
    at <enso_frp::stream::WeakNode<Def> as enso_frp::stream::WeakEventConsumer<T>>::on_event_if_exists::h523aa51918d39be4 (http://localhost:8080/pkg-opt.wasm:wasm-function[35202]:0x1a9347a)
    at <enso_frp::stream::OwnedStream<Out> as enso_frp::stream::EventEmitter>::emit_event::he9e3a4940042f77f (http://localhost:8080/pkg-opt.wasm:wasm-function[3654]:0xc130cc)
    at <enso_frp::stream::WeakNode<Def> as enso_frp::stream::WeakEventConsumer<T>>::on_event_if_exists::h62e3964a604b9122 (http://localhost:8080/pkg-opt.wasm:wasm-function[43793]:0x1c1f99d)
    at <enso_frp::stream::OwnedStream<Out> as enso_frp::stream::EventEmitter>::emit_event::he9e3a4940042f77f (http://localhost:8080/pkg-opt.wasm:wasm-function[3654]:0xc13088)
    at <enso_frp::stream::WeakNode<Def> as enso_frp::stream::WeakEventConsumer<T>>::on_event_if_exists::h523aa51918d39be4 (http://localhost:8080/pkg-opt.wasm:wasm-function[35202]:0x1a9347a)
    at <enso_frp::stream::OwnedStream<Out> as enso_frp::stream::EventEmitter>::emit_event::he9e3a4940042f77f (http://localhost:8080/pkg-opt.wasm:wasm-function[3654]:0xc13088)
    at <enso_frp::stream::WeakNode<Def> as enso_frp::stream::WeakEventConsumer<T>>::on_event_if_exists::h62e3964a604b9122 (http://localhost:8080/pkg-opt.wasm:wasm-function[43793]:0x1c1f99d)
    at <enso_frp::stream::OwnedStream<Out> as enso_frp::stream::EventEmitter>::emit_event::he9e3a4940042f77f (http://localhost:8080/pkg-opt.wasm:wasm-function[3654]:0xc13088)
    at <enso_frp::stream::WeakNode<Def> as enso_frp::stream::WeakEventConsumer<T>>::on_event_if_exists::he5db79135bb02f3f (http://localhost:8080/pkg-opt.wasm:wasm-function[38959]:0x1b48ae1)
    at <enso_frp::stream::OwnedStream<Out> as enso_frp::stream::EventEmitter>::emit_event::ha3ee661c3d310e0c (http://localhost:8080/pkg-opt.wasm:wasm-function[3847]:0xc62647)
    at <enso_frp::stream::WeakNode<Def> as enso_frp::stream::WeakEventConsumer<T>>::on_event_if_exists::h660d5dfd386bc5ad (http://localhost:8080/pkg-opt.wasm:wasm-function[35306]:0x1a988dc)
    at <enso_frp::stream::OwnedStream<Out> as enso_frp::stream::EventEmitter>::emit_event::h42c21cb188124536 (http://localhost:8080/pkg-opt.wasm:wasm-function[3792]:0xc4baa7)
    at <enso_frp::stream::WeakNode<Def> as enso_frp::stream::WeakEventConsumer<T>>::on_event_if_exists::h013eadffd310e522 (http://localhost:8080/pkg-opt.wasm:wasm-function[43432]:0x1c1069a)
    at <enso_frp::stream::OwnedStream<Out> as enso_frp::stream::EventEmitter>::emit_event::h42c21cb188124536 (http://localhost:8080/pkg-opt.wasm:wasm-function[3792]:0xc4baa7)
    at <enso_frp::stream::WeakNode<Def> as enso_frp::stream::WeakEventConsumer<T>>::on_event_if_exists::hb19accc2989aa6b3 (http://localhost:8080/pkg-opt.wasm:wasm-function[40062]:0x1b7c97c)
    at <enso_frp::stream::OwnedStream<Out> as enso_frp::stream::EventEmitter>::emit_event::he9e3a4940042f77f (http://localhost:8080/pkg-opt.wasm:wasm-function[3654]:0xc130cc)
    at <enso_frp::stream::WeakNode<Def> as enso_frp::stream::WeakEventConsumer<T>>::on_event_if_exists::h54a6cd455f3655d0 (http://localhost:8080/pkg-opt.wasm:wasm-function[25653]:0x1848b9a)
    at <enso_frp::stream::WeakNode<Def> as enso_frp::stream::WeakEventConsumer<T>>::on_event_if_exists::hae0c7230af26e3bb (http://localhost:8080/pkg-opt.wasm:wasm-function[2571]:0xa32625)
    at <enso_frp::stream::OwnedStream<Out> as enso_frp::stream::EventEmitter>::emit_event::ha14346811be68cd9 (http://localhost:8080/pkg-opt.wasm:wasm-function[2928]:0xada7cc)
    at enso_gui::presenter::searcher::provider::Component::provide_new_list::hf973466c39923dbf (http://localhost:8080/pkg-opt.wasm:wasm-function[5918]:0xf15a11)
    at <enso_frp::stream::WeakNode<Def> as enso_frp::stream::WeakEventConsumer<T>>::on_event_if_exists::h4435d65e4f9cff7e (http://localhost:8080/pkg-opt.wasm:wasm-function[8807]:0x1162501)
    at <enso_frp::stream::OwnedStream<Out> as enso_frp::stream::EventEmitter>::emit_event::he87e24c74a9dd9fa (http://localhost:8080/pkg-opt.wasm:wasm-function[4239]:0xd00f32)
    at <core::option::Option<T> as enso_prelude::option::OptionOps>::for_each::hdb8e75fe32476ffa (http://localhost:8080/pkg-opt.wasm:wasm-function[38454]:0x1b30e28)
    at enso_prelude::channel::process_stream_with_handle::{{closure}}::h76d38eff01e5cee3 (http://localhost:8080/pkg-opt.wasm:wasm-function[10824]:0x12a8cd8)
    at futures_util::stream::stream::StreamExt::poll_next_unpin::hb25ca4c94ef58f57 (http://localhost:8080/pkg-opt.wasm:wasm-function[5293]:0xe69e10)
    at futures_executor::local_pool::LocalPool::poll_pool::h0eb54d56a1c03493 (http://localhost:8080/pkg-opt.wasm:wasm-function[42089]:0x1bd6ab6)
    at std::thread::local::LocalKey<T>::with::h74244bd3d0e28437 (http://localhost:8080/pkg-opt.wasm:wasm-function[29984]:0x1971432)
    at futures_executor::local_pool::LocalPool::run_until_stalled::h57cd22f87075d591 (http://localhost:8080/pkg-opt.wasm:wasm-function[43339]:0x1c0c8f6)
    at ensogl_core::animation::loops::create_callback_wrapper::{{closure}}::h8d1d55a533363134 (http://localhost:8080/pkg-opt.wasm:wasm-function[46760]:0x1c88e14)
    at alloc::vec::Vec<T,A>::retain_mut::h41fdc9831e8bb459 (http://localhost:8080/pkg-opt.wasm:wasm-function[13015]:0x13c9f52)
    at core::ops::function::FnOnce::call_once{{vtable.shim}}::h645eef22accb5021 (http://localhost:8080/pkg-opt.wasm:wasm-function[6895]:0xff6505)
    at enso_callback::Registry<F>::run_once_impl::ha42fde738f0868aa (http://localhost:8080/pkg-opt.wasm:wasm-function[6321]:0xf77256)
    at <enso_callback::Registry<dyn core::ops::function::FnOnce<()>+Output = ()> as enso_callback::traits::RegistryRunner0>::run_all::h4362bb34f9020517 (http://localhost:8080/pkg-opt.wasm:wasm-function[62618]:0x1d65c62)
    at enso_frp::microtasks::SchedulerData::run_all::hc2f24fce00b9c615 (http://localhost:8080/pkg-opt.wasm:wasm-function[9033]:0x11894ce)
    at enso_frp::microtasks::Scheduler::new::{{closure}}::{{closure}}::hf28aa2acadb53374 (http://localhost:8080/pkg-opt.wasm:wasm-function[46163]:0x1c78a24)
    at <dyn core::ops::function::FnMut<(A,)>+Output = R as wasm_bindgen::closure::WasmClosure>::describe::invoke::h9f85d360c3080d27 (http://localhost:8080/pkg-opt.wasm:wasm-function[58860]:0x1d56400)
    at __wbg_adapter_86 (eval at <anonymous> (http://localhost:8080/index.js:74:5709), <anonymous>:16981:8)
    at real (eval at <anonymous> (http://localhost:8080/index.js:74:5709), <anonymous>:16907:14)

```

**Comment by vitvakatu** [comment]
My fault, we don't check for the size of the `window` argument here: https://github.com/enso-org/enso/blob/develop/app/gui/src/controller/searcher/input.rs#L444

I didn't know it panics if the size is zero. I can make a fix later today.

<blockquote><img src="https://opengraph.githubassets.com/ac4d90eca69290786574d29791c7f80a3b98c6a5e01741f94c543d280cbb70c8/enso-org/enso" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">enso/input.rs at develop · enso-org/enso</a></strong></div><div>Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.</div></blockquote>

**Comment by vitvakatu** [comment]
Yep, in this case the context is present (`.Count_Distinct`), but it has no name ~~arguments~~ segments, as we pop the last one and are left with `.` only. I didn't consider such thing.

**Comment by vitvakatu** [comment]
Closed in #6120
